### ISSUE TITLE
string/array repetition

### ISSUE THREAD

**Issue by satyr** [satyr/coco#52]
```
:A  'x'    * 3  # 'xxx'
:B  "#{x}" * 3  # (_ref = "" + x) + _ref + _ref
:C  'x'    * y  # Array(-~y).join('x')
:D  [x]    * 3  # [x, x, x]
:E  [x]    * y  # ??
```

_E_ (and _C_ for perf) may have to use helper functions.

**Comment by michaelficarra** [comment]
Nice, I miss this from Ruby. With regard to D (and probably E), what is the expected compilation for `[x, y] * 3`? I would guess `[x, y, x, y, x, y]`. Also, what about `[x++] * 3`? `[x++, x, x]`?

**Comment by satyr** [comment]
> expected compilation for `[x, y] * 3`?

`[x, y, x, y, x, y]`

> what about `[x++] * 3`?

`[_ref = x++, _ref, _ref]`

**Comment by satyr** [comment]
closed by 1e37e12d1fd5356f15c997fea57bfa5bce5ead93; string/array literals can now be replicated using `*`
